flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/ResultPartitionTest.java

```diff
@@ -205,6 +216,66 @@ protected void testAddOnPartition(final ResultPartitionType pipelined)
 		}
 	}
 
+	@Test
+	public void testReleaseMemoryOnBlockingPartition() throws Exception {
+		testReleaseMemory(ResultPartitionType.BLOCKING);
+	}
+
+	@Test
+	public void testReleaseMemoryOnPipelinedPartition() throws Exception {
+		testReleaseMemory(ResultPartitionType.PIPELINED);
+	}
+
+	/**
+	 * Tests {@link ResultPartition#releaseMemory(int)} on a working partition.
+	 *
+	 * @param resultPartitionType the result partition type to set up
+	 */
+	private void testReleaseMemory(final ResultPartitionType resultPartitionType) throws Exception {
+		final int numBuffers = 10;
+		final NetworkEnvironment network = new NetworkEnvironment(
```

Comment:
Could you deduplicate this constructor with other use cases? It is used in tests in 5 other places, 2 of them are almost exactly the same as this one and three others further duplicate this:
```
			new LocalConnectionManager(),
			new ResultPartitionManager(),
			new TaskEventDispatcher(),
			new KvStateRegistry(),
			null,
			null,
			IOManager.IOMode.SYNC,
```
block.

Please either provide some more constructors to deduplicate this or provide a `NetworkEnvironmentBuilder` (or `public static class Builder` nested in `NetworkEnvironment`) that would handle those default values. I would personally prefer the builder, since it's better for the future.

Reply:
Yes, I agree with the option of providing a `NetworkEnvironmentBuilder` nested in `NetworkEnvironment`. 
But I think it is worth opening a separate JIRA for unifying all the related history usages, because it is not caused by this PR, and this PR would be based on that new JIRA. What do you think?

Reply:
Another option is we provide another simple constructor for `NetworkEnvironment` instead of `Builder`, which one do you prefer?

I reviewed all the current usages and the above deduplication can cover the existing usages in tests except for `StreamNetworkBenchmarkEnvironment` which provides additional `ConnectionManager`.